FiveCalls/FiveCalls/ImpactViewModel.swift

```diff
@@ -30,4 +30,77 @@ struct ImpactViewModel {
     var voicemailCount: Int {
         return logs.filter { $0.outcome == .voicemail }.count
     }
+    
+    var weeklyStreakCount: Int {
+        let logDates = logs.map { $0.date }
+        return ImpactViewModel.countWeeklyStreak(callDates: logDates, now: Date())
+    }
+    
+    static func countWeeklyStreak(callDates: [Date], now: Date) -> Int {
```

Comment:
I could easily see this being a separate type entirely.  It's pure data, no dependencies, so it's an ideal candidate for extraction.

Maybe something like:

```swift
struct StreakCalculator {
  let dates: [Date]
  let referenceDate: Date
  
  // initializer we get for free?

  var consecutiveWeeks: Int {
         // logic
  }

  var consecutiveDays: Int {
     // we don't need this yet, but it's a good way to validate we're on the right path,
     // that it would also support other intervals
  }
}

Usage would look like this:

```swift
let calc = StreakCalculator(dates: ..., referenceDate: Date())
calc.consecutiveWeeks
```

I'm not crazy about the naming, open for suggestions.